packages/components/icons/src/Icon.tsx

```diff
@@ -0,0 +1,116 @@
+import { css } from '@emotion/core';
+import React, { forwardRef } from 'react';
+import { Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  Simplify,
+} from '@contentful/f36-core';
+import type { ComponentType, ReactElement } from 'react';
+import tokens from '@contentful/f36-tokens';
+
+export const styles = {
+  large: css({
+    height: '32px',
+    width: '32px',
+  }),
+  medium: css({
+    height: '24px',
+    width: '24px',
+  }),
+  negative: css({
+    fill: tokens.colorNegative,
+  }),
+  positive: css({
+    fill: tokens.colorPositive,
+  }),
+  root: css({
+    display: 'inline-block',
+    fill: tokens.colorBlueBase,
+  }),
+  secondary: css({
+    fill: tokens.colorContrastMid,
+  }),
+  small: css({
+    height: '18px',
+    width: '18px',
+  }),
+  tiny: css({
+    height: '16px',
+    width: '16px',
+  }),
+  trimmed: css({
+    width: 'auto',
+  }),
+  warning: css({
+    fill: tokens.colorWarning,
+  }),
+  white: css({
+    fill: tokens.colorWhite,
+  }),
+};
+
+type AsOrChildren =
+  | {
+      as: ComponentType;
+      children?: never;
+    }
+  | {
+      as?: never;
+      children: ReactElement | ReactElement[];
+    };
+
+type IconInternalProps = {
+  name?: string;
+  size?: 'large' | 'medium' | 'small' | 'tiny';
+  variant?: ComponentVariant;
+  viewBox?: `${number} ${number} ${number} ${number}`;
+} & AsOrChildren;
+
+export type IconProps = Simplify<
+  IconInternalProps & CommonProps & React.SVGAttributes<SVGSVGElement>
+>;
+
+export const Icon = forwardRef<SVGSVGElement, IconProps>(function Icon(
+  {
+    as,
```

Comment:
Will it be too confusing if we call this prop `as` even though it works differently than other `as` props? It doesn't take a string like `"div"`, only a component.

Reply:
I don't think it's confusing, it's merely a narrower subset of the previous type. The same will be with `Heading` component, for example, not all original values of `as` will be excepted, but only `h1`, `h2`, etc...